File: core/src/main/java/pcb/uwu/units/quantity/ScalarUnit.java
package pcb.uwu.units.quantity;

import pcb.uwu.core.BaseUnit;
import pcb.uwu.core.UnitCounter;

import java.util.function.Function;

public class ScalarUnit extends BaseUnit {

	public static final ScalarUnit SCALAR = new ScalarUnit();

	private final UnitCounter unitCounter;

	private ScalarUnit() {
		super(0, "", "", "", Function.identity(), Function.identity());

		this.unitCounter = new UnitCounter();
	}

	@Override
	public Class<? extends BaseUnit> getBaseUnitType() {
		return null;
	}

	@Override
	public UnitCounter getUnitCounter() {
		return new UnitCounter(); // todo change immutable singleton
	}
}

File: core/src/main/java/pcb/uwu/unit/base/ScalarUnit.kt
package pcb.uwu.unit.base

import pcb.uwu.core.BaseUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.utils.identity


object ScalarUnit
    : BaseUnit(precedence = 0,
               symbol = "",
               singularName = "",
               pluralName = "",
               translationToCanonical = identity(),
               translationFromCanonical = identity())
{
    override val unitCounter
        get() = UnitCounter()
    override val baseUnitType = ScalarUnit::class.java
}